samples/core/Miscellaneous/NewInEFCore7/DocumentsContext.cs

```diff
@@ -0,0 +1,656 @@
+                        builder => builder.HasOne<Person>().WithMany().OnDelete(DeleteBehavior.Cascade),
+                        builder => builder.HasOne<Book>().WithMany().OnDelete(DeleteBehavior.ClientCascade),
+                        joinTypeBuilder =>
+                        {
+                            if (UseStoredProcedures)
+                            {
+                                joinTypeBuilder.InsertUsingStoredProcedure(
+                                    "BookPerson_Insert",
+                                    storedProcedureBuilder =>
+                                    {
+                                        storedProcedureBuilder.HasParameter("AuthorsId");
+                                        storedProcedureBuilder.HasParameter("PublishedWorksId");
+                                    });
+
+                                joinTypeBuilder.DeleteUsingStoredProcedure(
+                                    "BookPerson_Delete",
+                                    storedProcedureBuilder =>
+                                    {
+                                        storedProcedureBuilder.HasParameter("AuthorsId");
+                                        storedProcedureBuilder.HasParameter("PublishedWorksId");
+                                    });
+                            }
+                        });
+            });
+
+        base.OnModelCreating(modelBuilder);
+    }
+}
+
+public class TptDocumentsContext : DocumentsContext
+{
+    public override MappingStrategy MappingStrategy => MappingStrategy.Tpt;
+
+    protected override void OnModelCreating(ModelBuilder modelBuilder)
+    {
+        modelBuilder.Entity<Document>().UseTptMappingStrategy();
+
+        modelBuilder.Entity<Document>(
+            entityTypeBuilder =>
+            {
+                if (UseStoredProcedures)
+                {
+                    entityTypeBuilder.InsertUsingStoredProcedure(
+                        "Document_Insert",
+                        storedProcedureBuilder =>
+                        {
+                            storedProcedureBuilder.HasParameter(document => document.Title);
+                            storedProcedureBuilder.HasParameter(document => document.NumberOfPages);
+                            storedProcedureBuilder.HasParameter(document => document.PublicationDate);
+                            storedProcedureBuilder.HasParameter(document => document.CoverArt);
+                            storedProcedureBuilder.HasResultColumn(document => document.Id);
+                            storedProcedureBuilder.HasResultColumn(document => document.FirstRecordedOn);
+                            storedProcedureBuilder.HasResultColumn(document => document.RetrievedOn);
+                            storedProcedureBuilder.HasResultColumn(document => document.RowVersion);
+                        });
+
+                    entityTypeBuilder.UpdateUsingStoredProcedure(
```

Comment:
This mapping is off: Document_Update accepts all columns for the entire hierarchy (so suitable for TPH), but is here being used for TPT... So the parameter list below doesn't correspond to the actual parameters accepted by the sproc.

Reply:
The TPT table is this:

```sql
      CREATE TABLE [Documents] (
          [Id] int NOT NULL IDENTITY,
          [RowVersion] rowversion NULL,
          [Title] nvarchar(max) NOT NULL,
          [NumberOfPages] int NOT NULL,
          [PublicationDate] datetime2 NOT NULL,
          [CoverArt] varbinary(max) NULL,
          [FirstRecordedOn] datetime2 NOT NULL DEFAULT (getutcdate()),
          [RetrievedOn] AS getutcdate(),
          CONSTRAINT [PK_Documents] PRIMARY KEY ([Id])
```

Which parameters are now correct?

Reply:
Sorry, I think I missed that you have different versions of the same sproc (Document_Update) based on which inheritance strategy is being used (maybe consider calling the sproc Document_Tpt_Update). I'll look again.